test/unittest/reporters/reporter/reporter_test.go

```diff
@@ -98,11 +98,82 @@ func TestReportWorkflow(t *testing.T) {
 
 	executionId, _ := uuid.Parse("6ba7b810-9dad-11d1-80b4-00c04fd430c8")
 
-	mock_ds_reporter.On("ReportWorkflow", executionId, playbook).Return(nil)
-	reporter.ReportWorkflow(executionId, playbook)
+	mock_ds_reporter.On("ReportWorkflowStart", executionId, playbook).Return(nil)
+	reporter.ReportWorkflowStart(executionId, playbook)
+
+	mock_ds_reporter.AssertExpectations(t)
+}
+
+func TestReportWorkflowEnd(t *testing.T) {
+	mock_ds_reporter := mock_reporter.Mock_Downstream_Reporter{}
+	reporter := reporter.New([]ds_reporter.IDownStreamReporter{&mock_ds_reporter})
+
+	expectedCommand := cacao.Command{
+		Type:    "ssh",
+		Command: "ssh ls -la",
+	}
+
+	expectedVariables := cacao.Variable{
+		Type:  "string",
+		Name:  "var1",
+		Value: "testing",
+	}
+
+	step1 := cacao.Step{
+		Type:          "action",
+		ID:            "action--test",
+		Name:          "ssh-tests",
+		StepVariables: cacao.NewVariables(expectedVariables),
+		Commands:      []cacao.Command{expectedCommand},
+		Cases:         map[string]string{},
+		OnCompletion:  "end--test",
+		Agent:         "agent1",
+		Targets:       []string{"target1"},
+	}
+
+	end := cacao.Step{
+		Type: "end",
+		ID:   "end--test",
+		Name: "end step",
+	}
+
+	expectedAuth := cacao.AuthenticationInformation{
+		Name: "user",
+		ID:   "auth1",
+	}
+
+	expectedTarget := cacao.AgentTarget{
+		Name:               "sometarget",
+		AuthInfoIdentifier: "auth1",
+		ID:                 "target1",
+	}
+
+	expectedAgent := cacao.AgentTarget{
+		Type: "soarca",
+		Name: "soarca-ssh",
+	}
+
+	playbook := cacao.Playbook{
+		ID:                            "test",
+		Type:                          "test",
+		Name:                          "ssh-test",
+		WorkflowStart:                 step1.ID,
+		AuthenticationInfoDefinitions: map[string]cacao.AuthenticationInformation{"id": expectedAuth},
+		AgentDefinitions:              map[string]cacao.AgentTarget{"agent1": expectedAgent},
+		TargetDefinitions:             map[string]cacao.AgentTarget{"target1": expectedTarget},
+
+		Workflow: map[string]cacao.Step{step1.ID: step1, end.ID: end},
+	}
+
+	executionId, _ := uuid.Parse("6ba7b810-9dad-11d1-80b4-00c04fd430c8")
+
+	mock_ds_reporter.On("ReportWorkflowEnd", executionId, playbook, nil).Return(nil)
+	reporter.ReportWorkflowEnd(executionId, playbook, nil)
+
+	mock_ds_reporter.AssertExpectations(t)
 }
 
```

Comment:
add a test to assert more then one reporter is called (add two downstream reporter mocks and assert they are called)

Reply:
The is more of an addition to this test, we should have done that with the reporter pull request but lets add it now